tokio/src/sync/watch.rs

```diff
@@ -365,9 +364,52 @@ impl<T> Receiver<T> {
     ///
     /// [`changed`]: Receiver::changed
     pub fn borrow_and_update(&mut self) -> Ref<'_, T> {
+        self.borrow_and_update_if_changed().0
+    }
+
+    /// Returns a reference to the most recently sent value with an *updated*
+    /// flag indicating if that value has been detected as *changed* and marked
+    /// as seen.
+    ///
+    /// This method marks the current value as seen. Subsequent calls to [`changed`]
+    /// will not return immediately until the producer has modified the shared value again.
+    ///
+    /// Outstanding borrows hold a read lock. This means that long lived borrows
+    /// could cause the send half to block. It is recommended to keep the borrow
+    /// as short lived as possible.
+    ///
+    /// The priority policy of the lock is dependent on the underlying lock
+    /// implementation, and this type does not guarantee that any particular policy
+    /// will be used. In particular, a producer which is waiting to acquire the lock
+    /// in `send` might or might not block concurrent calls to `borrow`, e.g.:
+    ///
+    /// <details><summary>Potential deadlock example</summary>
+    ///
+    /// ```text
+    /// // Task 1 (on thread A)                       |  // Task 2 (on thread B)
+    /// let (_ref1, _updated1) =                      |
+    ///         rx1.borrow_and_update_if_changed();   |
+    ///                                               |  // will block
+    ///                                               |  let _ = tx.send(());
+    /// // may deadlock                               |
+    /// let (_ref2, _updated2) =                      |
+    ///         rx2.borrow_and_update_if_changed();   |
+    /// ```
+    /// </details>
+    ///
+    /// [`changed`]: Receiver::changed
+    pub fn borrow_and_update_if_changed(&mut self) -> (Ref<'_, T>, bool) {
```

Comment:
That seems like a good way to go around it.

Reply:
Good point! I was already experimenting with such a design in my external wrapper crate. Just wanted to avoid introducing new, named types.

I didn't consider the fact that this solutions enables us to extend the current API without any breaking changes, which make it even more appealing :)

Reply:
I was blind :see_no_evil: Your proposal turned out as the perfect solution that only affects `borrow_and_update()` when focusing on the `updated`  aspect and not considering the more general `changed` aspect: https://github.com/tokio-rs/tokio/pull/4758/commits/63c9987da30bf5256eeedd62ff1d60580d06009b

Reply:
I noticed that commits will be squashed before merging, so I added the changes as a new commit.